.github/workflows/build-and-push-docker-hub.yaml

```
@@ -1,52 +1,341 @@
+
+      - name: Compute Docker image tags
+        id: set-tags
+        env:
+          INPUT_IMAGE_TAGS: ${{ inputs.image-tags }}
+        run: |
+          set -euo pipefail
+          TAGS="$INPUT_IMAGE_TAGS"
+          IFS=',' read -ra ELEMENTS <<< "$TAGS"
+          TAGS_FIXED=""
+          for ELEMENT in "${ELEMENTS[@]}"; do
+            ELEMENT_TRIMMED="$(echo "$ELEMENT" | xargs)"
+            [[ -z "$ELEMENT_TRIMMED" ]] && continue
+            TAGS_FIXED+="${IMAGE_FULL_NAME}:$ELEMENT_TRIMMED,"
+          done
+          TAGS_FIXED=${TAGS_FIXED%,}
+          if [[ -z "$TAGS_FIXED" ]]; then
+            echo "No valid tags were computed" >&2
+            exit 1
+          fi
+          echo "tags_fixed=$TAGS_FIXED" >> "$GITHUB_ENV"
+
+      - name: Determine target platforms
+        id: platforms
+        env:
+          RAW_PLATFORMS: ${{ inputs.platforms }}
+        run: |
+          set -euo pipefail
+          RAW="${RAW_PLATFORMS:-linux/amd64}"
+          RAW="${RAW//$'\r'/}"
+          RAW="${RAW//$'\n'/,}"
+          RAW="${RAW// /}"
+          PRIMARY=""
+          NORMALIZED=""
+          IFS=',' read -ra CANDIDATES <<< "$RAW"
+          for TOKEN in "${CANDIDATES[@]}"; do
+            TOKEN="${TOKEN//[$'\t\r\n ']}"
+            [[ -z "$TOKEN" ]] && continue
+            NORMALIZED+="${TOKEN},"
+            if [[ -z "$PRIMARY" ]]; then
+              PRIMARY="$TOKEN"
+            fi
+          done
+          NORMALIZED="${NORMALIZED%,}"
+          if [[ -z "$NORMALIZED" ]]; then
+            echo "No valid platforms were provided" >&2
+            exit 1
+          fi
+          echo "platforms=$NORMALIZED" >> "$GITHUB_OUTPUT"
+          echo "primary=$PRIMARY" >> "$GITHUB_OUTPUT"
+
+      - name: Prepare export directory
+        run: mkdir -p build/runtime
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@e468171a9de216ec08956ac3ada2f0791b6bd435 # v3.11.1
+
+      - name: Enable containerd image store
+        uses: depot/use-containerd-snapshotter-action@2e6f905d28e7ca519b216e2219d00545537e9f46 # v1.0.0
+
+      - name: Log in to Docker registry
+        uses: docker/login-action@5e57cd118135c172c3672efd75eb46360885c0ef # v3.6.0
+        with:
+          registry: ${{ env.REGISTRY_HOST }}
+          username: ${{ secrets.dockerhub_username }}
+          password: ${{ secrets.dockerhub_password }}
+
+      - name: Build & Push (SBOM + provenance)
+        id: build
+        uses: docker/build-push-action@263435318d21b8e681c14492fe198d362a7d2c83 # v6.18.0
+        with:
+          file: Dockerfile
+          context: .
+          push: true
+          platforms: ${{ steps.platforms.outputs.platforms }}
+          tags: ${{ env.tags_fixed }}
+          sbom: true
+          provenance: mode=max
+          outputs: |
+            type=registry,rewrite-timestamp=true,force-compression=true
+
+      - name: Build export stage (local artifacts)
+        id: build_export
```

Does this reuse the artifacts from the previous step? If so, rename this for clarity (and remove the `id` which is unused below):
```suggestion
      - name: Extract artifacts from export stage
```

If it doesn't, why not?